cpmpy/solvers/ortools.py

```diff
@@ -472,17 +472,37 @@ def _post_constraint(self, cpm_expr, reifiable=False):
                 return self.ort_model.AddCumulative(intervals, demand, cap)
             elif cpm_expr.name == "circuit":
                 # ortools has a constraint over the arcs, so we need to create these
-                # when using an objective over arcs, using these vars direclty is recommended
+                # when using an objective over arcs, using these vars directly is recommended
                 # (see PCTSP-path model in the future)
                 x = cpm_expr.args
                 N = len(x)
                 arcvars = boolvar(shape=(N,N))
                 # post channeling constraints from int to bool
                 self += [b == (x[i] == j) for (i,j),b in np.ndenumerate(arcvars)]
                 # post the global constraint
-                # when posting arcs on diagonal (i==j), it would do subcircuit
+                # when posting arcs on diagonal (i==j), it would do subcircuit (see subcircuit)
                 ort_arcs = [(i,j,self.solver_var(b)) for (i,j),b in np.ndenumerate(arcvars) if i != j]
                 return self.ort_model.AddCircuit(ort_arcs)
+            elif cpm_expr.name == "subcircuit":
+                x = cpm_expr.args
+                N = len(x)
+                arcvars = boolvar(shape=(N,N))
+                # post channeling constraints from int to bool
+                self += [b == (x[i] == j) for (i,j),b in np.ndenumerate(arcvars)]
+                # post the global constraint
+                # posting arcs on diagonal (i==j) allows for subcircuits
+                ort_arcs = [(i,j,self.solver_var(b)) for (i,j),b in np.ndenumerate(arcvars)] # Allows for empty subcircuits
+                return self.ort_model.AddCircuit(ort_arcs)
+            elif cpm_expr.name == "subcircuitwithstart":
```

Comment:
No, because SubCircuitWithStart decomposes into SubCircuit and a constraint requiring that the start_index is in the subcircuit and thus does not self-loop. And SubCircuit posts all pairwise node-node links (also all self-loops) in the graph to the solver. Whilst for ortools, the api allows to just never post the link between start_index and itself in the first place. So less `ort_arcs` posted and one constraint less.

Reply:
ORTools actually supports SubCircuitWithStart and doesn't need the decomposition though SubCircuit